lib/CTownHandler.cpp

```diff
@@ -1068,6 +1068,8 @@ CFaction * CTownHandler::loadFromJson(const std::string & scope, const JsonNode
 	
 	auto preferUndergound = source["preferUndergroundPlacement"];
 	faction->preferUndergroundPlacement = preferUndergound.isNull() ? false : preferUndergound.Bool();
+	auto isSpecial = source["special"];
+	faction->special = isSpecial.isNull() ? false : isSpecial.Bool();
```

Comment:
No need to check for null. JsonNode will return default value if key is not preset (so false for bools)
```suggestion
	faction->special = source["special"].Bool();
```